BTusingPreorderInorder.java

```diff
@@ -0,0 +1,58 @@
+//Time Complexity: O(n) 
```

Comment:
TC won't be O(n) because you are searching for the index in each recursive call. At the worst, this would run in polynomial time.